core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -557,8 +557,7 @@ private boolean moveShard(NodeSorter sorter, ShardRouting shardRouting, ModelNod
                 if (currentNode != sourceNode) {
                     RoutingNode target = currentNode.getRoutingNode();
                     Decision allocationDecision = allocation.deciders().canAllocate(shardRouting, target, allocation);
-                    Decision rebalanceDecision = allocation.deciders().canRebalance(shardRouting, allocation);
-                    if (allocationDecision.type() == Type.YES && rebalanceDecision.type() == Type.YES) { // TODO maybe we can respect throttling here too?
+                    if (allocationDecision.type() == Type.YES) { // TODO maybe we can respect throttling here too?
```

Comment:
can we add a comment here why we don't check "canRebalance"? (maybe link to this PR?)